src/socket.rs

```diff
@@ -118,14 +118,18 @@ fn addr2raw_v4(addr: &SocketAddrV4) -> (SocketAddrCRepr, c::socklen_t) {
             sin_family: c::AF_INET as c::sa_family_t,
             sin_port: addr.port().to_be(),
             sin_addr,
+            #[cfg(target_os = "haiku")]
+            sin_zero: [0; 24],
+            #[cfg(not(target_os = "haiku"))]
```

Comment:
Sorry to pick a style nit like this, but would you mind swapping the order on these, so the more common not-haiku comes first in the order?